frontend/src/actions/transfers/subsidized-transfer.ts

```diff
@@ -33,6 +33,22 @@ export class SubsidizedTransfer extends Transfer {
     this.sourceChain.requestManagerAddress = this.originalRequestManagerAddress;
   }
 
+  async withdraw(provider: IEthereumProvider): Promise<void> {
+    if (this.feeSubAddress) {
+      this.sourceChain.requestManagerAddress = this.feeSubAddress;
+    }
+    await super.withdraw(provider);
+    this.sourceChain.requestManagerAddress = this.originalRequestManagerAddress;
+  }
```

Comment:
Is there something like Python's `try...finally` in JS?
What happens if `withdraw` throws an exception that is caught in one of the callers?

Reply:
Thats a very good point Ivan. I will wrap it inside a try/catch block.